crates/ruff_python_parser/tests/snapshots/invalid_syntax@statements__with__ambiguous_lpar_with_items.py.snap

```diff
@@ -1429,16 +1432,45 @@ Module(
  9 | with (item := 10 as f): ...
 10 | with (item1, item2 := 10 as f): ...
 11 | with (x for x in range(10), item): ...
-   |       ^^^^^^^^^^^^^^^^^^^^ Syntax Error: Unparenthesized generator expression cannot be used here
+   |                           ^ Syntax Error: Expected ')', found ','
 12 | with (item, x for x in range(10)): ...
    |
 
 
    |
+ 9 | with (item := 10 as f): ...
 10 | with (item1, item2 := 10 as f): ...
 11 | with (x for x in range(10), item): ...
+   |                                 ^ Syntax Error: Expected ',', found ')'
 12 | with (item, x for x in range(10)): ...
-   |             ^^^^^^^^^^^^^^^^^^^^ Syntax Error: Unparenthesized generator expression cannot be used here
```

Comment:
The problem here is that the speculative parsing failed so we fallback to considering parenthesized expression. The terminator token for with-items parsing with the kind being parenthesized expression is `:`, so it'll expect a `,` after a with item (`item`) but not at the end (`:`).

I think I want to improve the logic of parsing comma-separated list from `(element comma) (element comma) ...` to `(element) (comma element) (comma element) ...`.